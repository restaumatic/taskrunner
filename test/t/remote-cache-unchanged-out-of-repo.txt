# no toplevel
# s3

export TASKRUNNER_SAVE_REMOTE_CACHE=1

mkdir a

(
  cd a
  echo foo > input.txt
  git init -q
  git add input.txt
)

cp -r a b

mkdir -p out_a out_b

go() {
  export out_dir="$1"
  # Note: we want separate workdir for separate repos
  TASKRUNNER_STATE_DIRECTORY="$(pwd)" TASKRUNNER_ROOT_DIRECTORY="$(pwd)" taskrunner -n mytask bash -e -c '
    snapshot input.txt --cache-root "${out_dir}" --outputs "${out_dir}/output.txt"
    echo "Expensive computation"
    rev input.txt > "${out_dir}/output.txt"
  '
}

(
  cd a
  go ../out_a
  echo "Output in a is: $(cat ../out_a/output.txt)"
)

(
  cd b
  go ../out_b # note: should not compute again
  echo "Output in b is: $(cat ../out_b/output.txt)"
)
