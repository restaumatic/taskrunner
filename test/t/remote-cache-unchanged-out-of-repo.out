-- output:
[mytask] info | Inputs changed, running task
[mytask] stdout | Expensive computation
[mytask] info | success
Output in a is: oof
[mytask] info | Found remote cache mytask-6b36308659163b577fd1c832107dc46ca3aa659b.tar.zst, restoring
[mytask] info | Restored from remote cache
Output in b is: oof
